packages/block-library/src/comments-query-loop/edit.js

```diff
@@ -3,47 +3,61 @@
  */
 import {
 	BlockControls,
-	InspectorControls,
 	useBlockProps,
 	useInnerBlocksProps,
 } from '@wordpress/block-editor';
-import { SelectControl } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import QueryToolbar from './toolbar';
+import CommentsToolbar from './edit/toolbar';
+import CommentsInspectorControls from './edit/comments-inspector-controls';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
 
 const TEMPLATE = [ [ 'core/comment-template' ] ];
 
 export default function CommentsQueryLoopEdit( { attributes, setAttributes } ) {
-	const { perPage, tagName: TagName } = attributes;
+	const { tagName: TagName, order } = attributes;
```

Comment:
Shouldn't we also handle the default `perPage` assignment in `Comments Query` and not in the `Comments Template` block as [is done now](https://github.com/WordPress/gutenberg/pull/37297/files#diff-885985bee90d7d51f079b99d59d8562cba40b4a6f64258e86fc81b0dc0f11b87R150)? This way it will always be filled by `context` and the errors in the `Comments Query` controls will go away.

Reply:
If the values are not set for `perPage`, and/or `order` we want them to be read from the site settings. Otherwise, when we save explicitly the current site settings in the serialized block, any further global change won't be reflected in the block. I guess it's up to debate which approach is better because both have pros and cons, but at least that was my thinking when I worked on the initial implementation for `perPage`.